sunscreen/src/types/zkp/gadgets/binary.rs

```diff
@@ -23,6 +33,12 @@ impl Gadget for ToUInt {
             return Err(ZkpError::gadget_error("Cannot create 0-bit uint."));
         }
 
+        if self.n > 512 {
+            return Err(ZkpError::gadget_error(
```

Comment:
These seems redundant since it is checked in the constructor, but I suppose it doesn't hurt

Reply:
Removed